### PR TITLE
Polkadot v0.9.38 upgrade

### LINKED ISSUE #1118: Upgrade to Polkadot Release v0.9.38
The Parity team has published new version of Polkadot: `v0.9.38`

See:
- https://github.com/paritytech/polkadot/releases/tag/v0.9.38
- https://forum.polkadot.network/t/polkadot-release-analysis-v0-9-38

## Upgrade Priority

See [Polkadot Release Page](https://github.com/paritytech/polkadot/releases/tag/v0.9.38)

## Host Functions

See [Polkadot Release Page](https://github.com/paritytech/polkadot/releases/tag/v0.9.38)

## Database Migrations

See [Polkadot Release Page](https://github.com/paritytech/polkadot/releases/tag/v0.9.38)

## Runtime Migrations

See [Polkadot Release Page](https://github.com/paritytech/polkadot/releases/tag/v0.9.38)

### PR DESCRIPTION
**Pull request by mattheworris**
# Goal
The goal of this PR is upgrade Polkadot to v0.9.38
See: https://github.com/paritytech/polkadot/releases/tag/v0.9.38

Closes #1118

# Discussion

# Checklist
- [x] Chain spec updated
- [x] Weights updated
- [x] Tests
  - [x] make ci-local: passsing all tests
  - [x] make upgrade-local: passing against local relay network

### CODE REVIEW

**Comment by rlaferla** [comment]
Interval sealing is not working in this PR.

**Review by rlaferla — APPROVED** [review]
Tested:

1. Create MSA extrinsic
- Instant sealing passed
- Manual sealing passed
- Interval sealing passed
2. Create user handle passed
3. Governance council proposal passed
4. Unit and integration tests passed

File: .github/workflows/verify-pr-commit.yml [hunk 1]
```diff
@@ -209,8 +209,14 @@ jobs:
     steps:
       - name: Check Out Repo
         uses: actions/checkout@v3
-      - name: Build Rust Docs
-        run: RUSTDOCFLAGS="--enable-index-page --check -Zunstable-options" cargo doc --no-deps --features frequency
+        # cargo doc requires a toolchain version ^nightly-2022-11-15 because of a bug that causes a panic
+        # When the toolchain is updated to ^nightly-2023-06-01, this can be removed
+      - name: Install Rust Nightly-2023-06-01 and Build Docs
+        run: |
+          export HOME="/root"
```
Comment (mattheworris) [inline .github/workflows/verify-pr-commit.yml:216]: I don't know why this `export` was required, but it fixes an issue where rust was complaining about rustup not being installed in `/github/home/.cargo`.

Comment (wilwade) [reply .github/workflows/verify-pr-commit.yml:216]: I think this is fine for now as a temp fix.

@demisx Any thoughts?

File: node/service/src/service.rs [hunk 2]
```diff
@@ -216,10 +216,8 @@ async fn start_node_impl(
 		hwbench.clone(),
 	)
 	.await
-	.map_err(|e| match e {
-		RelayChainError::ServiceError(polkadot_service::Error::Sub(x)) => x,
-		s => s.to_string().into(),
-	})?;
+	// REVIEW: from cumulus
+	.map_err(|e| sc_service::Error::Application(Box::new(e) as Box<_>))?;
```
Comment (wilwade) [inline node/service/src/service.rs:220]: Comment?

File: pallets/frequency-tx-payment/src/lib.rs [hunk 3]
```diff
@@ -22,6 +22,7 @@
 // Substrate macros are tripping the clippy::expect_used lint.
 #![allow(clippy::expect_used)]
 #![cfg_attr(not(feature = "std"), no_std)]
+#![feature(rustdoc_missing_doc_code_examples)]
```
Comment (wilwade) [inline pallets/frequency-tx-payment/src/lib.rs:25]: Curious why these were added.

File: runtime/common/src/weights/block_weights.rs [hunk 4]
```diff
@@ -1,8 +1,8 @@
 //! THIS FILE WAS AUTO-GENERATED USING THE SUBSTRATE BENCHMARK CLI VERSION 4.0.0-dev
-//! DATE: 2023-04-05 (Y/M/D)
-//! HOSTNAME: `benchmark-runner-p5rt6-vk9q8`, CPU: `Intel(R) Xeon(R) Platinum 8375C CPU @ 2.90GHz`
+//! DATE: 2023-06-02 (Y/M/D)
+//! HOSTNAME: `benchmark-runner-qpqf8-4dvlm`, CPU: `Intel(R) Xeon(R) Platinum 8375C CPU @ 2.90GHz`
 //!
-//! SHORT-NAME: `block`, LONG-NAME: `BlockExecution`, RUNTIME: `Frequency Local Testnet`
+//! SHORT-NAME: `block`, LONG-NAME: `BlockExecution`, RUNTIME: `Frequency Development (No Relay)`
```
Comment (wilwade) [inline runtime/common/src/weights/block_weights.rs:5]: I don't think it is due to this PR, but I'm not sure this should be `No Relay` for overhead weights, as it doesn't include the parachain interaction overhead weights.

Talk with @JoeCap08055 who setup the overhead benchmarks and see what he thinks. The benchmarks look the same below, so perhaps this doesn't matter.

Non-blocking for this PR, but needs to be resolved before the next release.

File: runtime/frequency/src/lib.rs [hunk 5]
```diff
@@ -679,8 +679,9 @@ impl pallet_timestamp::Config for Runtime {
 // the descriptions of these configs.
 impl pallet_authorship::Config for Runtime {
 	type FindAuthor = pallet_session::FindAccountFromAuthorIndex<Self, Aura>;
-	type UncleGenerations = AuthorshipUncleGenerations;
-	type FilterUncle = ();
+	// REVIEW: Were these traits removed from the pallet?
```
Comment (wilwade) [inline runtime/frequency/src/lib.rs:682]: Comment?

File: runtime/frequency/src/lib.rs [hunk 6]
```diff
@@ -1330,6 +1331,14 @@ impl_runtime_apis! {
...
 			TransactionPayment::query_fee_details(uxt, len)
 		}
+		// REVIEW: substrate
+		fn query_weight_to_fee(weight: Weight) -> Balance {
+			TransactionPayment::weight_to_fee(weight)
+		}
+		// REVIEW: substrate
+		fn query_length_to_fee(len: u32) -> Balance {
+			TransactionPayment::length_to_fee(len)
+		}
```
Comment (wilwade) [inline runtime/frequency/src/lib.rs:1341]: Remaining comments

**Review by wilwade — CHANGES_REQUESTED** [review]
Few things to check on and questions, but looks good.

File: pallets/frequency-tx-payment/src/lib.rs (hunk 3 above)
Comment (mattheworris) [reply pallets/frequency-tx-payment/src/lib.rs:25]: `make lint` was complaining about `unknown lint`. Adding the feature seemed like the least intrusive fix.

**Review by wilwade — APPROVED** [review]

**Pull request merged**